llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -13170,15 +13177,23 @@ static SDValue tryToFoldExtOfLoad(SelectionDAG &DAG, DAGCombiner &Combiner,
        !TLI.isLoadExtLegal(ExtLoadType, VT, N0.getValueType())))
```

Comment:
What about this legality check? Could we end up producing an illegal sextload?

Reply:
Good catch. Maybe I'll just pre-scan for signed setccs instead of trying to integrate it into the existing scan so I can do it early.